src/Modal.jsx

```diff
@@ -15,6 +18,22 @@ class Modal extends React.Component {
 		super(props);
 		this.onDismiss = this.onDismiss.bind(this);
 		this.onKeyDown = this.onKeyDown.bind(this);
+		this.getModalPosition = this.getModalPosition.bind(this);
+
+		this.state = {
+			topPosition: DEFAULT_MARGIN_TOP // matches default margin-top in CSS
+		};
+
+		if (!this.props.fullscreen && typeof window.matchMedia != 'undefined') {
+			this.mediaQuery = window.matchMedia(MEDIA_QUERIES.medium);
+
+			this.handleMediaChange = mq => {
```

Comment:
`componentDidMount` currently has the same structure - break it out into a named function that can be called from both places.